PCbuild/find_python.bat

```diff
@@ -34,10 +34,10 @@
 @if exist "%_Py_EXTERNALS_DIR%\pythonx86\tools\python.exe" (set PYTHON="%_Py_EXTERNALS_DIR%\pythonx86\tools\python.exe") & (set _Py_Python_Source=found in externals directory) & goto :found
 
 @rem If HOST_PYTHON is recent enough, use that
-@if NOT "%HOST_PYTHON%"=="" @%HOST_PYTHON% -Ec "import sys; assert sys.version_info[:2] >= (3, 6)" >nul 2>nul && (set PYTHON="%HOST_PYTHON%") && (set _Py_Python_Source=found as HOST_PYTHON) && goto :found
+@if NOT "%HOST_PYTHON%"=="" @%HOST_PYTHON% -Ec "import sys; assert sys.version_info[:2] >= (3, 8)" >nul 2>nul && (set PYTHON="%HOST_PYTHON%") && (set _Py_Python_Source=found as HOST_PYTHON) && goto :found
 
 @rem If py.exe finds a recent enough version, use that one
-@for %%p in (3.8 3.7 3.6) do @py -%%p -EV >nul 2>&1 && (set PYTHON=py -%%p) && (set _Py_Python_Source=found %%p with py.exe) && goto :found
+@for %%p in (3.9 3.8) do @py -%%p -EV >nul 2>&1 && (set PYTHON=py -%%p) && (set _Py_Python_Source=found %%p with py.exe) && goto :found
```

Comment:
Bumping the versions here works, but it is worth considering changing the construct somewhat to find a minimum usable version, just like for HOST_PYTHON above.

Reply:
Yup I agree. I realised that this would have to be manually updated on every new Python version released. And while that's only a once-a-year affair, it'd be nice to do what you said and forget about the script :).

Reply:
Using `for /F "tokens=2 delims=-" %%p in ('py -0 2^>nul') do @py -%%p ...` will allow for the same type of test that HOST_PYTHON uses

Reply:
Thank you for this suggestion. I didn't know ``for`` was this versatile!